kv/tikv/storage/server.go

```diff
@@ -0,0 +1,178 @@
+package storage
+
+import (
+	"context"
+	"github.com/pingcap-incubator/tinykv/kv/pd"
+	"github.com/pingcap-incubator/tinykv/kv/tikv/dbreader"
+	"github.com/pingcap-incubator/tinykv/kv/tikv/inner_server"
+	"github.com/pingcap-incubator/tinykv/kv/tikv/storage/kvstore"
+	"github.com/pingcap-incubator/tinykv/kv/tikv/storage/latches"
+	"github.com/pingcap-incubator/tinykv/proto/pkg/coprocessor"
+	"github.com/pingcap-incubator/tinykv/proto/pkg/kvrpcpb"
+	"github.com/pingcap-incubator/tinykv/proto/pkg/tikvpb"
+)
+
+// InnerServer represents the internal-facing server part of TinyKV, it handles sending and receiving from other
+// TinyKV nodes. As part of that responsibility, it also reads and writes data to disk (or semi-permanent memory).
+type InnerServer interface {
+	Start(pdClient pd.Client) error
+	Stop() error
+	Write(ctx *kvrpcpb.Context, batch []inner_server.Modify) error
+	Reader(ctx *kvrpcpb.Context) (dbreader.DBReader, error)
+	Raft(stream tikvpb.Tikv_RaftServer) error
+	Snapshot(stream tikvpb.Tikv_SnapshotServer) error
+}
+
+var _ tikvpb.TikvServer = new(Server)
+
+// Server is a TinyKV server, it 'faces outwards', sending and receiving messages from clients such as TinySQL.
+type Server struct {
+	innerServer InnerServer
+	Latches     *latches.Latches
+}
+
+func NewServer(innerServer InnerServer) *Server {
+	return &Server{
+		innerServer: innerServer,
+		Latches:     latches.NewLatches(),
+	}
+}
+
+func (server *Server) Run(cmd Command) (interface{}, error) {
```

Comment:
I'm not sure it is necessary to extract the command the logic still after the scheduler is removed.
Writing individual logic for every transaction API seems is simpler with just a little boilerplate code, see
https://github.com/ngaut/unistore/blob/master/tikv/server.go#L287
Though both look fine for me.